tests/python/direct/test_tutorial.py

```diff
@@ -572,3 +581,563 @@ def test_tutorial_id_model_reshape_analysis():
     # should be mapped too.
     for tv2_id, tv3_id in zip(tv2.get_loop_domain(), tv3.get_loop_domain()):
         assert exact_graph.disjoint_val_sets().strict_are_mapped(tv2_id, tv3_id)
+
+
+@pytest.mark.skipif(
+    is_pre_hopper(), reason="Only supported on Hopper and newer devices."
+)
+def test_tutorial_basic_tma_example1(nvfuser_direct_test):
+    """
+    This tutorial uses copy kernels to demonstrate how to schedule TMA.
+    Please note that this is not a guide on how to use TMA to achieve SOL.
+    Instead, it is a demonstration on the degree of freedoms we have in a
+    TMA schedule and how a schedule is translated into generated code in the
+    kernel. I also want the readers to focus on the schedule of TMA. The
+    other parts of the kernel that is scheduled is not important here.
+    Indeed, I picked a random, valid schedule. For the example about TMA
+    load, please focus on the schedule of the shared memory tensor. For the
+    example about TMA store, please focus on the allocation domain of the
+    shared memory tensor and the fusion output.
+    """
+
+    # In this example, we treat the fusion as 1D, which is similar to how we
+    # generally schedule pointwise fusions. We use a single 1D TMA instruction to
+    # load the entire CTA tile to shared memory.
+    # CTA tile size = TMA tile size = 256
+    with FusionDefinition() as fd:
+        input = fd.define_tensor(shape=[-1, -1, -1], contiguity=[True, True, True])
+        output = fd.ops.set(input)
+        fd.add_output(output)
+
+        smem_cache = input.cache_after(LoadStoreOpType.tma)
+        smem_cache.set_memory_type(MemoryType.shared)
+
+        # For TMA load, both the shared memory layout and the loop nest and
+        # parallelization of TMA are specified by the consumer: smem_cache
+
+        # Step 1: define TMA domain
+        # We want to treat the entire tensor as 1D so define the TMA domain as
+        # [I0*I1*I2]
+        smem_cache.merge(0, 1)
+        smem_cache.merge(0, 1)
+        # Note that the TMA domain only exists in people's mind, there is no need to
+        # set anything here.
+
+        # Step 2: define box
+        smem_cache.split(0, 256)
+        # [I0*I1*I2/256, 256]
+        # partitioned IterDomain: I0*I1*I2
+        # coordinate IterDomain: I0*I1*I2/256
+        # box IterDomain: 256
+
+        # Step 3: define tile
+        # We use dense tile here, so tile == box. Nothing to do here.
+
+        # Step 4: schedule the shared memory tensor
+        # By default, the allocation domain is the logical domain, which is already
+        # in good shape for this case.
+
+        # Step 5: schedule the consumer tensor
+        smem_cache.axis(0).parallelize(ParallelType.grid_x)
+        smem_cache.axis(1).parallelize(ParallelType.tma)
+        # [BIDx, Bulk]
```

Comment:
nit: Should we rename `ParallelType::Bulk` into `ParallelType::TMA` to be consistent in naming?